Marlin/stepper.cpp

```diff
@@ -490,14 +490,24 @@ ISR(TIMER1_COMPA_vect) {
       // TEST_ENDSTOP: test the old and the current status of an endstop
       #define TEST_ENDSTOP(ENDSTOP) (TEST(current_endstop_bits, ENDSTOP) && TEST(old_endstop_bits, ENDSTOP))
 
-      #define UPDATE_ENDSTOP(AXIS,MINMAX) \
-        SET_ENDSTOP_BIT(AXIS, MINMAX); \
-        if (TEST_ENDSTOP(_ENDSTOP(AXIS, MINMAX))  && (current_block->steps[_AXIS(AXIS)] > 0)) { \
-          endstops_trigsteps[_AXIS(AXIS)] = count_position[_AXIS(AXIS)]; \
-          _ENDSTOP_HIT(AXIS); \
-          step_events_completed = current_block->step_event_count; \
-        }
-      
+      #ifndef COREXY
+        #define UPDATE_ENDSTOP(AXIS,MINMAX) \
+          SET_ENDSTOP_BIT(AXIS, MINMAX); \
+          if (TEST_ENDSTOP(_ENDSTOP(AXIS, MINMAX))  && (current_block->steps[_AXIS(AXIS)] > 0)) { \
+            endstops_trigsteps[_AXIS(AXIS)] = count_position[_AXIS(AXIS)]; \
+            _ENDSTOP_HIT(AXIS); \
+            step_events_completed = current_block->step_event_count; \
+          }	  
+      #else // CoreXY checks for axis movement, not steps.
+	    #define UPDATE_ENDSTOP(AXIS,MINMAX) \
+	  	  SET_ENDSTOP_BIT(AXIS, MINMAX); \
+		  if (TEST_ENDSTOP(_ENDSTOP(AXIS, MINMAX))  && (current_block->axis_movement[_AXIS(AXIS)] != 0)) { \
```

Comment:
Oh, I was unaware of that. I see on stackoverflow that comparing float to 0 is unrealiable depending on the compiler, as for some   -0.0f != +0.0f, is that what you are pointing out?

Maybe I can try something like:
```C
abs(current_block->axis_movement[_AXIS(AXIS)]) > 0
```